runtime/mgmt/mgmtinit.c

```diff
@@ -31,3 +33,106 @@ JNI_OnLoad(JavaVM *vm, void *reserved)
 {
 	return JNI_VERSION_1_8;
 }
+
+/* Convert path to platform encoding.
+ *
+ * @return NULL on error, otherwise return converted string
+ */
+static const char*
+translatePathToPlatform(JNIEnv *env, const char* pathUTF)
+{
+	char *result;
+	int32_t bufferLength = 0;
+	int32_t resultLength = 0;
+
+	PORT_ACCESS_FROM_ENV(env);
+
+	bufferLength = 	j9str_convert(J9STR_CODE_MUTF8, J9STR_CODE_PLATFORM_RAW, pathUTF, strlen(pathUTF), NULL, 0);
+
+	/* get the size of the platform string */
+	if (bufferLength < 0 ) {
+		return NULL;  /* error occurred */
+	} else {
+		bufferLength += MAX_STRING_TERMINATOR_LENGTH;
+	}
+
+	result = (char*)j9mem_allocate_memory(bufferLength, OMRMEM_CATEGORY_PORT_LIBRARY);
+	if (NULL == result)  {
+		return NULL;
+	}
+
+	resultLength = j9str_convert(J9STR_CODE_MUTF8, J9STR_CODE_PLATFORM_RAW, pathUTF, strlen(pathUTF), result, bufferLength);
+
+	/* do the conversion */
+	if ((resultLength < 0) ||  (resultLength + MAX_STRING_TERMINATOR_LENGTH != bufferLength)) {
+		j9mem_free_memory(result);
+		return NULL; /* error occurred */
+	} else {
+		/* add terminator */
+		memset(result + resultLength, 0, MAX_STRING_TERMINATOR_LENGTH);
+		return (const char*)result;
+	}
+}
+
+static void throwNativeOutOfMemError(JNIEnv* env) {
+	J9VMThread *currentThread = (J9VMThread *)env;
+	J9JavaVM *vm = currentThread->javaVM;
+	J9InternalVMFunctions *vmFuncs = vm->internalVMFunctions;
+	vmFuncs->internalEnterVMFromJNI(currentThread);
+	vmFuncs->setNativeOutOfMemoryError(currentThread, 0, 0);
+	vmFuncs->internalReleaseVMAccess(currentThread);
+}
+
+/**
+ * Determines whether file path has user permissions only.
+ *
+ * @param path file path
+ * @return Returns false on Windows and true if the file has only user permissions,
+ * false otherwise on other operation systems.
+ */
+JNIEXPORT jboolean JNICALL 
+Java_sun_management_FileSystemImpl_isAccessUserOnly0(JNIEnv *env, jclass c, jstring path)
+{
+	const char* pathUTF;
+	jboolean result = JNI_FALSE;
+
+	PORT_ACCESS_FROM_ENV(env);
+
+	pathUTF = (*env)->GetStringUTFChars(env, path, NULL);
+	if (NULL == pathUTF) {
+error:
+		throwNativeOutOfMemError(env);
+	} else {
+		struct J9FileStat buf;
+		I_32 status;
+		const char* pathConvert;
+
+#if !defined(WIN32) && !defined(WIN64)
+		pathConvert = (char*)translatePathToPlatform(env, pathUTF);
+		if (NULL == pathConvert) {
+			(*env)->ReleaseStringUTFChars(env, path, pathUTF);
+			goto error;
+		}
+#else
+		pathConvert = pathUTF;
+#endif /* WIN */
+
+		/* retrieve file information */
+		status = j9file_stat(pathUTF, 0, &buf);
+
+		if (0 == status) {
+			/* if any permissions are set other than user, fail */
+			if (!buf.perm.isGroupWriteable && !buf.perm.isGroupReadable && !buf.perm.isOtherWriteable && !buf.perm.isOtherReadable) {
+				result = JNI_TRUE;
+			}
+		} else {
+			/* failed to retrieve statistics */
+			goto error;
+		}
+
+		(*env)->ReleaseStringUTFChars(env, path, pathUTF);
+		j9mem_free_memory((void*)pathConvert);
```

Comment:
if `pathConvert` is assigned `pathUTF` as an initial value as suggest above, this will need a `if (pathConvert != pathUTF)` check.... and that will need to be done before calling ReleaseStringUTFChars on pathUTF.